### PR TITLE
Refs [TASK][36092] Tạo folder data và Koin module

### PR DESCRIPTION
**Pull request by trinq-2471**
## Related Tickets
- [#Task 36092: Tạo folder data và Koin module](https://edu-redmine.sun-asterisk.vn/issues/36092)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/moviedb_54/data/model/MovieResult.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+data class MovieResult(
+    val id: Int? = null,
+    val title: String? = null,
+
+    @SerializedName("poster_path")
+    val posterPath: String? = null,
+
+    @SerializedName("vote_average")
+    val voteAverage: Double? = null
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/data/model/MovieResult.kt]:
```suggestion
    val id: Int? = null,
    val title: String? = null,
    @SerializedName("poster_path")
    val posterPath: String? = null,
    @SerializedName("vote_average")
    val voteAverage: Double? = null
```

File: app/src/main/java/com/sun/moviedb_54/data/source/remote/RetrofitClient.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.sun.moviedb_54.data.source.remote
+
+import com.sun.moviedb_54.ultis.Constant.BASE_URL
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class RetrofitClient(factory: GsonConverterFactory, client: OkHttpClient) {
+    private var baseRetrofit = Retrofit.Builder()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/data/source/remote/RetrofitClient.kt]:
```suggestion
class RetrofitClient(factory: GsonConverterFactory, client: OkHttpClient) {

    private var baseRetrofit = Retrofit.Builder()
```

File: app/src/main/java/com/sun/moviedb_54/extensions/LiveDataExt.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+operator fun <T> MutableLiveData<MutableList<T>>.plusAssign(item: T) {
+    val value = this.value ?: mutableListOf()
+    value.add(item)
+    this.value = value
+}
+
+operator fun <T> MutableLiveData<MutableList<T>>.plusAssign(item: MutableList<T>) {
+    val value = this.value ?: mutableListOf()
+    value.addAll(item)
+    this.postValue(value)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/extensions/LiveDataExt.kt:14]: scope function

File: app/src/main/java/com/sun/moviedb_54/ultis/MyApplication.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import com.sun.moviedb_54.module.apiModule
+import com.sun.moviedb_54.module.repositoryModule
+import com.sun.moviedb_54.module.retrofitModule
+import com.sun.moviedb_54.module.viewModelModule
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.core.context.startKoin
+
+class MyApplication : Application() {
+    override fun onCreate() {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/ultis/MyApplication.kt]:
```suggestion
class MyApplication : Application() {

    override fun onCreate() {
```

File: app/src/main/java/com/sun/moviedb_54/extensions/LiveDataExt.kt (hunk 3 above)
Comment (trinq-2471) [reply app/src/main/java/com/sun/moviedb_54/extensions/LiveDataExt.kt:14]: chổ này mình thấy viết vậy ngắn nhất rồi ạ, m thử dùng scope thì thấy nó viết còn phức tạp hơn ạ

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**